content/5_Plat/HLD.mdx

```diff
@@ -252,7 +252,137 @@ set problems where $\Theta(Q\sqrt N\log N)$ is intended ...
 	</Resource>
 </Resources>
 
-<IncompleteSection />
+
+<!-- <IncompleteSection /> -->
+
+The heavy-light decomposition algorithm is versatile and can be combined with other data structures like segment trees, binary indexed trees, and range minimum query data structures to solve a wide range of complex tree-related problems.
+
+## Template
+
+<LanguageSection>
+<CPPSection>
+```cpp
+
+// BeginCodeSnip{Segment Tree}
+struct Segtree {
+	int n;
+	vector<int> st;
+
+	void init(int a, const vector<int> &v) {
+		n = a;
+		st.assign(2 * n, 0);
+		for (int i = n; i < 2 * n; ++i) { st[i] = v[i - n]; }
+		for (int i = n - 1; i > 0; --i) {
+			st[i] = max(st[2 * i], st[2 * i + 1]);
+		}
+	}
+
+	int query(int a, int b) {
+		int s = 0;
+		a += n - 1;
+		b += n - 1;
+		while (a <= b) {
+			if (a % 2 == 1) { s = max(s, st[a++]); }
+			if (b % 2 == 0) { s = max(s, st[b--]); }
+			a /= 2, b /= 2;
+		}
+		return s;
+	}
+
+	void update(int p, int v) {
+		for (st[p += n - 1] = v; p > 1; p /= 2) {
+			st[p / 2] = max(st[p], st[p ^ 1]);
+		}
+	}
+};
+// EndCodeSnip
+
+struct HLD {
```

Comment:
although you have this template here, none of the internal sols + content in the module uses it. maybe it would be better to have a commented ver. of the form that you are using.

Reply:
commented version?

Reply:
like instead of the template, just write in the same format your code and other sols use and add comments.

Reply:
added.